products/web_analytics/dags/cache_warming.py

```diff
@@ -0,0 +1,236 @@
+                    'web_overview_query',
+                    'web_vitals_path_breakdown_query'
+                )
+        ) AS sub
+        GROUP BY
+            team_id,
+            query_json_raw,
+            normalized_query_hash
+        HAVING query_count >= %(minimum_query_count)s
+        ORDER BY
+            query_count DESC
+        """,
+        {"team_id": team_id, "days": days, "minimum_query_count": minimum_query_count},
+    )
+
+    return [
+        {
+            "team_id": result[0],
+            "query_json": json.loads(result[1]),
+            "query_count": result[2],
+            "normalized_query_hash": result[3],
+        }
+        for result in results
+    ]
+
+
+@dagster.op()
+def get_teams_for_warming_op(
+    context: dagster.OpExecutionContext, posthoganalytics: PostHogAnalyticsResource
+) -> list[int]:
+    team_ids = teams_enabled_for_web_analytics_cache_warming()
+
+    context.log.info(f"Found {len(team_ids)} teams for cache warming")
+    context.add_output_metadata({"team_count": len(team_ids), "team_ids": str(team_ids)})
+    return team_ids
+
+
+@dagster.op
+def get_queries_for_teams_op(
+    context: dagster.OpExecutionContext,
+    team_ids: list[int],
+) -> list[dict]:
+    days = get_instance_setting("WEB_ANALYTICS_WARMING_DAYS")
+    minimum_query_count = get_instance_setting("WEB_ANALYTICS_WARMING_MIN_QUERY_COUNT")
+
+    all_queries = []
+    for team_id in team_ids:
+        queries = queries_to_keep_fresh(context, team_id, days=days, minimum_query_count=minimum_query_count)
+
+        context.log.info(f"Loading {len(queries)} frequent web analytics queries for team {team_id}")
+
+        STALE_WEB_QUERIES_GAUGE.labels(team_id=team_id).set(len(queries))
+        all_queries.extend(queries)
+
+    context.log.info(f"Found {len(all_queries)} total queries to warm")
+    context.add_output_metadata({"query_count": len(all_queries), "team_count": len(team_ids)})
+    return all_queries
+
+
+@dagster.op(retry_policy=cache_warming_retry_policy)
+def warm_queries_op(context: dagster.OpExecutionContext, queries: list[dict]) -> None:
+    warmed_count = 0
+    cached_count = 0
+
+    for query_info in queries:
+        team_id = query_info["team_id"]
+        query_json = query_info["query_json"]
+        normalized_query_hash = query_info["normalized_query_hash"]
+
+        try:
+            team = Team.objects.get(pk=team_id)
+        except Team.DoesNotExist:
+            context.log.warning(f"Team {team_id} not found, skipping")
+            continue
+
+        tag_queries(team_id=team_id, trigger="webAnalyticsQueryWarming", feature=Feature.CACHE_WARMUP)
+
+        try:
+            results = process_query_dict(
+                team,
+                query_json,
+                limit_context=LimitContext.QUERY_ASYNC,
+                execution_mode=ExecutionMode.RECENT_CACHE_CALCULATE_BLOCKING_IF_STALE,
+            )
+
+            is_cached = getattr(results, "is_cached", False)
```

Comment:
Can we know if it was cached on that execution? I assume the execution mode guarantees that but I wonder how would we would know if we're firing already-cached-queries requests

Reply:
Adding something else here, but if we only enable this for us, it's not a problem right now, but we should make it clearer in comments or tests: Can we skip queries that we now are already cached? 

Right now is: let's rely on just doing it for queries that happened at least N times and if we run the same query again, it should hit the cached version, which I assume it is ok to testing it out but we should have some better guarantees later on to:

- Not run too many warmings per team
- Not keep trying queries that failed or ran out of memory
- Be mindful of backpressure